anjali/sb-pt-cont/3.js

```diff
@@ -0,0 +1,105 @@
+// https://levelbuilder-studio.code.org/s/sb-pt-cont/lessons/1/levels/3
+/*
+Make two sprites with different costumes and different locations. Use the pin on the location block to move sprites to different locations.
+*/
+if (!validationProps.successCriteria) {
+  validationProps.successCriteria = {
+    twoSprites: false,
+    differentLocations: false,
+    differentCostumes: false
+  };
+}
+
+//Helper variables
+var spriteIds = getSpriteIdsInUse();
+
+//Count Sprites
+validationProps.successCriteria.twoSprites = spriteIds.length>=2;
+
+if (World.frameCount == 1) {
+  // Check sprite locations
+  var uniqueStartingSpriteLocations = [];
+  for (var i=0; i<spriteIds.length; i++) {
+    var coords = [getProp({id: spriteIds[i]}, "x"), getProp({id: spriteIds[i]}, "y")];
+    console.log("coords for " + spriteIds[i] + ": " + coords);
+    var noDuplicateCoords = true;
+    for (var j=0; j<uniqueStartingSpriteLocations.length; j++) {
+      if ((coords[0] == uniqueStartingSpriteLocations[j][0]) &&
+          (coords[1] == uniqueStartingSpriteLocations[j][1])){
+        noDuplicateCoords = false;
+        break;
+      }
+    }
+    if (!noDuplicateCoords) {
+      break;
+    } else {
+      uniqueStartingSpriteLocations.push(coords);
+    }
+  }
+  if (spriteIds.length == uniqueStartingSpriteLocations.length) {
+    validationProps.successCriteria.differentLocations = true;
+  }
+
+  // Check sprite costumes
+  var uniqueStartingSpriteCostumes = [];
+  for (var i=0; i<spriteIds.length; i++) {
+    var costume = getProp({id: spriteIds[i]}, "costume");
+    var noDuplicateCostumes = true;
+    for (var j=0; j<uniqueStartingSpriteCostumes.length; j++) {
+      if (costume == uniqueStartingSpriteCostumes[j]) {
+        noDuplicateCostumes = false;
+        break;
+      }
+    }
+    if (!noDuplicateCostumes) {
+      break;
+    } else {
+      uniqueStartingSpriteCostumes.push(costume);
+    }
+  }
+  if (spriteIds.length == uniqueStartingSpriteCostumes.length) {
+    validationProps.successCriteria.differentCostumes = true;
+  }
+}
+
+// Set success time if success
+if (validationProps.successCriteria.twoSprites &&
+    validationProps.successCriteria.differentLocations &&
+    validationProps.successCriteria.differentCostumes &&
+   !validationProps.successTime)
+{
+  validationProps.successTime = World.frameCount;
+}
+
+// Delay fail time (so student can observe the wrong animation)
+var failTime = 10;
```

Comment:
10 is still pretty fast to fail the level, can you check with Mike about these times?

Reply:
I changed them to failTime = 90 for now. We can ask Mike when he's back.